[flaky test] Test_Run_Positive_VolumeAttachMountUnmountDetach

<!-- Please only use this template for submitting reports about flaky tests or jobs (pass or fail with no underlying change in code) in Kubernetes CI -->

#### Which jobs are flaking:

pull-kubernetes-unit

#### Which test(s) are flaking:

`k8s.io/kubernetes/pkg/kubelet/volumemanager/reconciler`: `Test_Run_Positive_VolumeAttachMountUnmountDetach`

#### Testgrid link:

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/99874/pull-kubernetes-unit/1367990215217516544

#### Reason for failure:

```
E0306 00:22:58.676574   65580 nestedpendingoperations.go:301] Operation for "{volumeName:fake-plugin/fake-device1 podName: nodeName:mynodename}" failed. No retries permitted until 2021-03-06 00:22:59.17633852 +0000 UTC m=+1.109197490 (durationBeforeRetry 500ms). Error: "DetachVolume.Detach failed for volume \"volume-name\" (UniqueName: \"fake-plugin/fake-device1\") on node \"mynodename\" : Trying to detach volume \"fake-device1\" that is not attached to the node \"mynodename\""
    reconciler_test.go:347: 
        	Error Trace:	reconciler_test.go:347
        	Error:      	Received unexpected error:
        	            	No Detachers have expected DetachCallCount. Expected: <1>.
        	Test:       	Test_Run_Positive_VolumeAttachMountUnmountDetach
--- FAIL: Test_Run_Positive_VolumeAttachMountUnmountDetach (0.20s)
```

#### Anything else we need to know:
- links to go.k8s.io/triage appreciated
- links to specific failures in spyglass appreciated

<!-- Please see the deflaking doc (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/flaky-tests.md) for more guidance! -->

/sig storage

--- a/pkg/volume/testing/testing.go
+++ b/pkg/volume/testing/testing.go
@@ -1084,7 +1084,6 @@ func (fv *FakeVolume) GetUnmountDeviceCallCount() int {
 func (fv *FakeVolume) Detach(volumeName string, nodeName types.NodeName) error {
 	fv.Lock()
 	defer fv.Unlock()
-	fv.DetachCallCount++
 
 	node := string(nodeName)
 	volumeNodes, exist := fv.VolumesAttached[volumeName]
@@ -1092,6 +1091,7 @@ func (fv *FakeVolume) Detach(volumeName string, nodeName types.NodeName) error {
 		return fmt.Errorf("trying to detach volume %q that is not attached to the node %q", volumeName, node)
 	}
 
+	fv.DetachCallCount++
 	if nodeName == FailDetachNode {
 		return fmt.Errorf("fail to detach volume %q to node %q", volumeName, nodeName)
 	}

